### ISSUE TITLE
Identify mechanisms of each rare disease

### ISSUE THREAD

**Issue by bschilder** [neurogenomics/rare_disease_celltyping#52]
We can use the Monarch knowledge graph to extract info on _how_ each gene is linked to each rare disease phenotype; i.e. missense mutation, splicing mutations, disrupted promoter, methylation, etc.

**Comment by bschilder** [comment]
I mapped all variants in ClinVar to functional regions using `VariantAnnotation`:
https://github.com/neurogenomics/KGExplorer/blob/master/R/map_variants.R

I then plotted all ClinVar variants annotations by frequency, faceted by pathogenicity status. Interestingly, the vast majority are splicing mutations!

![clinvar_plot](https://github.com/neurogenomics/rare_disease_celltyping/assets/34280215/477f928f-6fb0-4aac-8fe5-0c7d04ece996)

This data also includes disease/phenotype-variant links. While the specific variant for a specific patient is going to vary from patient to patient, this will at least give us a sense of what kinds of variants mechanisms are at play for a given gene within a given disease.

**Comment by bschilder** [comment]
See here for a nice example of how ASOs can be used to treat rare conditions:
https://www.nature.com/articles/s41586-023-06277-0

**Comment by bschilder** [comment]
Actually, it seems that subsetting ClinVar by data source has a big impact.

## HPO-only
If we're targeting phenotypes, splice sites seems to be the most relevant overall. This is kind of confusing because (as you'll see) the disease-level filtering identifies coding mutations as the dominant mechanism.

![variants_hpo_subset](https://github.com/neurogenomics/rare_disease_celltyping/assets/34280215/d4075e4c-14bc-4f72-95fa-3f40eb7b1852)

## OMIM-only

However when you subset to only ClinVar variants associated with OMIM diseases, coding mutations become the dominant mechanism.
![variants_omim_subset](https://github.com/neurogenomics/rare_disease_celltyping/assets/34280215/a999ca49-5e20-47d3-bd52-1ad6dd4e75d0)

## OMIM + Orphanet

Coding mutations remain dominant when considering both OMIM and Orphanet togther.
![variants_omim-orphanet_subset](https://github.com/neurogenomics/rare_disease_celltyping/assets/34280215/587fc685-0ad3-4fda-9514-546b35775ed6)

# Conclusion

Exactly which phenotypes/diseases you consider can change the predominant variant-level mechanism. But for our purposes, what's actually important is not necessarily at the aggregate level, but as the phenotype target-specific level.

In other words, once we identify our gene therapy target, we can subset this data to a particular candidate gene for a specific phenotype and identify the most common types of variant-level mechanisms within that gene.
